### PR TITLE
feat: added auth pages to video library

### PR DESCRIPTION
**Pull request by kedark152**
### feat: added authentication pages to video library

Features added in this PR are:-

- added login/signup pages, used mock backend api functionality.
- created useAuth custom hook using useContext
- added login/logout button in navbar.
- added toast for auth notification.
- User now can log in or create new account through Sign up

Preview Link: https://deploy-preview-3--agroplay.netlify.app/

### CODE REVIEW

File: src/context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+const AuthProvider = ({ children }) => {
+  const [auth, setAuth] = useState({});
+  useEffect(() => {
+    (async () => {
+      const token = localStorage.getItem("token");
+      token
+        ? setAuth({ token: token, isLoggedIn: true })
+        : setAuth({ token: "", isLoggedIn: false });
+    })();
```
Comment (omjadhav85) [inline src/context/auth-context.js]: i dont think an async function is required here because as per my knowledge local storage calls are synchronous

**Review by omjadhav85 — APPROVED** [review]
Great work! Easy to read code and clean UI. Can implement suggested minor change. Else looks good to merge!

File: src/context/auth-context.js (hunk 1 above)
Comment (kedark152) [reply src/context/auth-context.js]: Thanks Omkar, will make the changes.

**Review by AnuragSahu11 — APPROVED** [review]
Good use of useRef , proper variable names, ready to merge

**Pull request merged**